### ISSUE TITLE
Mhap precompute jobs failed (canu 1.6)

### ISSUE THREAD

**Issue by Alxdu** [marbl/canu#1056]
First time canu user on Linux (Ubuntu). I am getting an error that terminates de-novo assembly of some fungal DNA reads (see below the pasted text for commands used and the canu output). 

I am not very experienced with coding or de-novo assembly of nanopore reads, so please keep that in mind. You will see I was trying to assemble a 34Mb genome with roughly 6.3 read coverage,and I thought that may take less time for an initial test run. I think I can manage to gather up to 13-15x read coverage for this genome, but I won't hit 20x coverage. (the final strategy was to combine illumina with nanopore, but for now I just want to see what the nanopore output looks like). 

Any ideas on what could trigger this mhap error? 
Thank you very much,
Alex

```
> canu \
> stopOnReadQuality=false \
> -p run13BC03 -d canu_assembly \
> genomeSize=34.0m \
> -nanopore-raw *.fastq

-- Canu 1.6
--
-- CITATIONS
--
-- Koren S, Walenz BP, Berlin K, Miller JR, Phillippy AM.
-- Canu: scalable and accurate long-read assembly via adaptive k-mer weighting and repeat separation.
-- Genome Res. 2017 May;27(5):722-736.
-- http://doi.org/10.1101/gr.215087.116
-- 
-- Read and contig alignments during correction, consensus and GFA building use:
--   Šošic M, Šikic M.
--   Edlib: a C/C ++ library for fast, exact sequence alignment using edit distance.
--   Bioinformatics. 2017 May 1;33(9):1394-1395.
--   http://doi.org/10.1093/bioinformatics/btw753
-- 
-- Overlaps are generated using:
--   Berlin K, et al.
--   Assembling large genomes with single-molecule sequencing and locality-sensitive hashing.
--   Nat Biotechnol. 2015 Jun;33(6):623-30.
--   http://doi.org/10.1038/nbt.3238
-- 
--   Myers EW, et al.
--   A Whole-Genome Assembly of Drosophila.
--   Science. 2000 Mar 24;287(5461):2196-204.
--   http://doi.org/10.1126/science.287.5461.2196
-- 
--   Li H.
--   Minimap and miniasm: fast mapping and de novo assembly for noisy long sequences.
--   Bioinformatics. 2016 Jul 15;32(14):2103-10.
--   http://doi.org/10.1093/bioinformatics/btw152
-- 
-- Corrected read consensus sequences are generated using an algorithm derived from FALCON-sense:
--   Chin CS, et al.
--   Phased diploid genome assembly with single-molecule real-time sequencing.
--   Nat Methods. 2016 Dec;13(12):1050-1054.
--   http://doi.org/10.1038/nmeth.4035
-- 
-- Contig consensus sequences are generated using an algorithm derived from pbdagcon:
--   Chin CS, et al.
--   Nonhybrid, finished microbial genome assemblies from long-read SMRT sequencing data.
--   Nat Methods. 2013 Jun;10(6):563-9
--   http://doi.org/10.1038/nmeth.2474
-- 
-- CONFIGURE CANU
--
-- Detected Java(TM) Runtime Environment '10.0.1' (from 'java').
-- Detected gnuplot version '5.2 patchlevel 2' (from 'gnuplot') and image format 'png'.
-- Detected 8 CPUs and 31 gigabytes of memory.
-- No grid engine detected, grid disabled.
--
--                            (tag)Concurrency
--                     (tag)Threads          |
--            (tag)Memory         |          |
--        (tag)         |         |          |  algorithm
--        -------  ------  --------   --------  -----------------------------
-- Local: meryl      8 GB    4 CPUs x   2 jobs  (k-mer counting)
-- Local: cormhap    6 GB    8 CPUs x   1 job   (overlap detection with mhap)
-- Local: obtovl     8 GB    8 CPUs x   1 job   (overlap detection)
-- Local: utgovl     8 GB    8 CPUs x   1 job   (overlap detection)
-- Local: cor        7 GB    2 CPUs x   4 jobs  (read correction)
-- Local: ovb        3 GB    1 CPU  x   8 jobs  (overlap store bucketizer)
-- Local: ovs        8 GB    1 CPU  x   8 jobs  (overlap store sorting)
-- Local: red        2 GB    4 CPUs x   2 jobs  (read error detection)
-- Local: oea        1 GB    1 CPU  x   8 jobs  (overlap error adjustment)
-- Local: bat       15 GB    4 CPUs x   2 jobs  (contig construction)
-- Local: cns       15 GB    4 CPUs x   2 jobs  (consensus)
-- Local: gfa        8 GB    4 CPUs x   2 jobs  (GFA alignment and processing)
--
-- Found Nanopore uncorrected reads in the input files.
--
-- Generating assembly 'run13BC03' in '/home/nanopore/Nanopore_data/20180730_1926_H_run13_fungi/barcode03/canu_assembly'
--
-- Parameters:
--
--  genomeSize        34000000
--
--  Overlap Generation Limits:
--    corOvlErrorRate 0.3200 ( 32.00%)
--    obtOvlErrorRate 0.1440 ( 14.40%)
--    utgOvlErrorRate 0.1440 ( 14.40%)
--
--  Overlap Processing Limits:
--    corErrorRate    0.5000 ( 50.00%)
--    obtErrorRate    0.1440 ( 14.40%)
--    utgErrorRate    0.1440 ( 14.40%)
--    cnsErrorRate    0.1920 ( 19.20%)
--
--
-- BEGIN CORRECTION
--
----------------------------------------
-- Starting command on Thu Aug 16 14:54:49 2018 with 202.135 GB free disk space

    cd correction
    /usr/lib/canu/gatekeeperCreate \
      -minlength 1000 \
      -o ./run13BC03.gkpStore.BUILDING \
      ./run13BC03.gkpStore.gkp \
    > ./run13BC03.gkpStore.BUILDING.err 2>&1

-- Finished on Thu Aug 16 14:54:50 2018 (1 second) with 202.055 GB free disk space
----------------------------------------
--
-- In gatekeeper store 'correction/run13BC03.gkpStore':
--   Found 59296 reads.
--   Found 222069969 bases (6.53 times coverage).
--
--   Read length histogram (one '*' equals 262.78 reads):
--        0    999      0 
--     1000   1999  18395 **********************************************************************
--     2000   2999  12199 **********************************************
--     3000   3999   8269 *******************************
--     4000   4999   5993 **********************
--     5000   5999   4353 ****************
--     6000   6999   3130 ***********
--     7000   7999   2227 ********
--     8000   8999   1585 ******
--     9000   9999   1048 ***
--    10000  10999    740 **
--    11000  11999    496 *
--    12000  12999    339 *
--    13000  13999    192 
--    14000  14999    125 
--    15000  15999     83 
--    16000  16999     47 
--    17000  17999     35 
--    18000  18999     12 
--    19000  19999     10 
--    20000  20999      7 
--    21000  21999      2 
--    22000  22999      1 
--    23000  23999      2 
--    24000  24999      2 
--    25000  25999      0 
--    26000  26999      1 
--    27000  27999      2 
--    28000  28999      0 
--    29000  29999      1 
-- Finished stage 'cor-gatekeeper', reset canuIteration.
-- Finished stage 'merylConfigure', reset canuIteration.
--
-- Running jobs.  First attempt out of 2.
----------------------------------------
-- Starting 'meryl' concurrent execution on Thu Aug 16 14:54:50 2018 with 202.053 GB free disk space (1 processes; 2 concurrently)

    cd correction/0-mercounts
    ./meryl.sh 1 > ./meryl.000001.out 2>&1

-- Finished on Thu Aug 16 14:55:14 2018 (24 seconds) with 201.442 GB free disk space
----------------------------------------
-- Meryl finished successfully.
-- Finished stage 'merylCheck', reset canuIteration.
--
--  16-mers                                                                                           Fraction
--    Occurrences   NumMers                                                                         Unique Total
--       1-     1 124495655 *******************************************************************--> 0.7997 0.5629
--       2-     2  17312802 ********************************************************************** 0.9110 0.7194
--       3-     4  10734889 *******************************************                            0.9568 0.8162
--       5-     7   2700487 **********                                                             0.9907 0.9192
--       8-    11    257247 *                                                                      0.9981 0.9531
--      12-    16     60405                                                                        0.9990 0.9596
--      17-    22     28884                                                                        0.9993 0.9628
--      23-    29     17462                                                                        0.9995 0.9651
--      30-    37     12724                                                                        0.9996 0.9671
--      38-    46      9827                                                                        0.9997 0.9690
--      47-    56      7618                                                                        0.9998 0.9708
--      57-    67      5844                                                                        0.9998 0.9725
--      68-    79      5055                                                                        0.9998 0.9741
--      80-    92      4786                                                                        0.9999 0.9758
--      93-   106      4630                                                                        0.9999 0.9777
--     107-   121      3879                                                                        0.9999 0.9797
--     122-   137      2548                                                                        1.0000 0.9817
--     138-   154      1329                                                                        1.0000 0.9831
--     155-   172       697                                                                        1.0000 0.9840
--     173-   191       409                                                                        1.0000 0.9845
--     192-   211       313                                                                        1.0000 0.9848
--     212-   232       261                                                                        1.0000 0.9851
--     233-   254       194                                                                        1.0000 0.9853
--     255-   277       189                                                                        1.0000 0.9855
--     278-   301       154                                                                        1.0000 0.9858
--     302-   326       119                                                                        1.0000 0.9860
--     327-   352       143                                                                        1.0000 0.9862
--     353-   379        91                                                                        1.0000 0.9864
--     380-   407        91                                                                        1.0000 0.9865
--     408-   436        72                                                                        1.0000 0.9867
--     437-   466        63                                                                        1.0000 0.9868
--     467-   497        60                                                                        1.0000 0.9869
--     498-   529        62                                                                        1.0000 0.9871
--     530-   562        53                                                                        1.0000 0.9872
--     563-   596        48                                                                        1.0000 0.9874
--     597-   631        54                                                                        1.0000 0.9875
--     632-   667        44                                                                        1.0000 0.9876
--     668-   704        35                                                                        1.0000 0.9878
--     705-   742        36                                                                        1.0000 0.9879
--     743-   781        29                                                                        1.0000 0.9880
--     782-   821        34                                                                        1.0000 0.9881
--
--       31085 (max occurrences)
--    96684874 (total mers, non-unique)
--    31174265 (distinct mers, non-unique)
--   124495655 (unique mers)
-- For mhap overlapping, set repeat k-mer threshold to 2211.
--
-- Found 221180529 16-mers; 155669920 distinct and 124495655 unique.  Largest count 31085.
-- Finished stage 'cor-meryl', reset canuIteration.
--
-- OVERLAPPER (mhap) (correction)
--
-- Set corMhapSensitivity=high based on read coverage of 6.
--
-- PARAMETERS: hashes=768, minMatches=2, threshold=0.78
--
-- Given 6 GB, can fit 9000 reads per block.
-- For 8 blocks, set stride to 2 blocks.
-- Logging partitioning to 'correction/1-overlapper/partitioning.log'.
-- Configured 7 mhap precompute jobs.
-- Configured 13 mhap overlap jobs.
-- Finished stage 'cor-mhapConfigure', reset canuIteration.
--
-- Running jobs.  First attempt out of 2.
----------------------------------------
-- Starting 'cormhap' concurrent execution on Thu Aug 16 14:55:15 2018 with 202.053 GB free disk space (7 processes; 1 concurrently)

    cd correction/1-overlapper
    ./precompute.sh 1 > ./precompute.000001.out 2>&1
    ./precompute.sh 2 > ./precompute.000002.out 2>&1
    ./precompute.sh 3 > ./precompute.000003.out 2>&1
    ./precompute.sh 4 > ./precompute.000004.out 2>&1
    ./precompute.sh 5 > ./precompute.000005.out 2>&1
    ./precompute.sh 6 > ./precompute.000006.out 2>&1
    ./precompute.sh 7 > ./precompute.000007.out 2>&1

-- Finished on Thu Aug 16 14:55:16 2018 (1 second) with 201.835 GB free disk space
----------------------------------------
--
-- Mhap precompute jobs failed, retry.
--   job correction/1-overlapper/blocks/000001.dat FAILED.
--   job correction/1-overlapper/blocks/000002.dat FAILED.
--   job correction/1-overlapper/blocks/000003.dat FAILED.
--   job correction/1-overlapper/blocks/000004.dat FAILED.
--   job correction/1-overlapper/blocks/000005.dat FAILED.
--   job correction/1-overlapper/blocks/000006.dat FAILED.
--   job correction/1-overlapper/blocks/000007.dat FAILED.
--
--
-- Running jobs.  Second attempt out of 2.
----------------------------------------
-- Starting 'cormhap' concurrent execution on Thu Aug 16 14:55:16 2018 with 201.835 GB free disk space (7 processes; 1 concurrently)

    cd correction/1-overlapper
    ./precompute.sh 1 > ./precompute.000001.out 2>&1
    ./precompute.sh 2 > ./precompute.000002.out 2>&1
    ./precompute.sh 3 > ./precompute.000003.out 2>&1
    ./precompute.sh 4 > ./precompute.000004.out 2>&1
    ./precompute.sh 5 > ./precompute.000005.out 2>&1
    ./precompute.sh 6 > ./precompute.000006.out 2>&1
    ./precompute.sh 7 > ./precompute.000007.out 2>&1

-- Finished on Thu Aug 16 14:55:17 2018 (1 second) with 201.835 GB free disk space
----------------------------------------
--
-- Mhap precompute jobs failed, tried 2 times, giving up.
--   job correction/1-overlapper/blocks/000001.dat FAILED.
--   job correction/1-overlapper/blocks/000002.dat FAILED.
--   job correction/1-overlapper/blocks/000003.dat FAILED.
--   job correction/1-overlapper/blocks/000004.dat FAILED.
--   job correction/1-overlapper/blocks/000005.dat FAILED.
--   job correction/1-overlapper/blocks/000006.dat FAILED.
--   job correction/1-overlapper/blocks/000007.dat FAILED.
--

ABORT:
ABORT: Canu 1.6
ABORT: Don't panic, but a mostly harmless error occurred and Canu stopped.
ABORT: Try restarting.  If that doesn't work, ask for help.
ABORT:
nanopore@nanopore-Precision-Tower-3620:~/Nanopore_data/20180730_1926_H_run13_fungi/barcode03$ 
```

**Comment by skoren** [comment]
Most likely an issue with your java install/version. One of the failed log files (```correction/1-overlapper/precompute.*.out```) should have more details (see #1055 for similar error). You're also on 1.6, upgrade to the latest release which is 1.7.1.

**Comment by Alxdu** [comment]
Thanks. Will try the java and canu updates and circle back to this data set. Thanks for the quick reply. Very appreciated

**Comment by skoren** [comment]
Any updates on this?

**Comment by skoren** [comment]
Idle, likely a JVM issue. Nothing to fix in Canu.

**Comment by Alxdu** [comment]
Hello,

I didn't find the time to fully troubleshoot the issue. I'll flag it again
if I hit any snags after installing the newer version of Canu

On Sep 25, 2018 9:40 AM, "Sergey Koren" <notifications@github.com> wrote:

Closed #1056 <https://github.com/marbl/canu/issues/1056>.

—
You are receiving this because you authored the thread.
Reply to this email directly, view it on GitHub
<https://github.com/marbl/canu/issues/1056#event-1865761309>, or mute the
thread

**Comment by jackchen129** [comment]
> Idle, likely a JVM issue. Nothing to fix in Canu.

Hi, I just ran into the same problem. And it seems I installed Canu 1.6.
I'm new to Linux. Could you please tell me how I can update it to 1.8 and what commands should I type in?
The Java version is 11.0.2. And how can I uninstall it and install Java 8?
Thanks.
